### ISSUE TITLE
Have you tried learning yoloe with Scratch?

### ISSUE THREAD

**Issue by ImJaewooChoi** [ultralytics/ultralytics#21428]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/orgs/ultralytics/discussions) and found no similar questions.

### Question

Have you tried learning yoloe with scracth?
It seems that it cannot be reimplemented. Nan appears in valid loss.

**Comment by UltralyticsAssistant** [comment]
👋 Hello @ImJaewooChoi, thank you for your interest in Ultralytics 🚀! We recommend a visit to the [Docs](https://docs.ultralytics.com/) for new users where you can find many [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage examples and where many of the most common questions may already be answered.

If this is a 🐛 Bug Report, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) to help us debug it.

If this is a custom training ❓ Question, please provide as much information as possible, including dataset image examples and training logs, and verify you are following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/).

Join the Ultralytics community where it suits you best. For real-time chat, head to [Discord](https://discord.com/invite/ultralytics) 🎧. Prefer in-depth discussions? Check out [Discourse](https://community.ultralytics.com/). Or dive into threads on our [Subreddit](https://www.reddit.com/r/Ultralytics/) to share knowledge with the community.

> ℹ️ This is an automated response, and an Ultralytics engineer will assist you here soon.

## Upgrade

Upgrade to the latest `ultralytics` package including all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/) to verify your issue is not already resolved in the latest version:

```bash
pip install -U ultralytics
```

## Environments

YOLO may be run in any of the following up-to-date verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda-zone)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

**Comment by glenn-jocher** [comment]
YOLOE training from scratch can be challenging and the NaN loss you're experiencing is likely due to training instability. According to the [YOLOE documentation](https://docs.ultralytics.com/models/yoloe/), the recommended approach is to use the `YOLOESegTrainerFromScratch` trainer class with specific hyperparameters like `lr0=2e-3`, `optimizer="AdamW"`, and `weight_decay=0.025`. The documentation also notes that Visual Prompt models should be fine-tuned based on trained-well Text Prompt models rather than starting completely from scratch. Try using the provided training configuration exactly as shown in the docs and consider starting from a pre-trained checkpoint if the issue persists.

**Comment by Snitro** [comment]
I encountered the same issue. All metrics became NaN after warmup.
Setting amp=False fixed the problem for me.

``` python
model.train(
    data=data,  # or data="yoloe_data.yaml" if using YAML file
    batch=128,
    epochs=30,
    close_mosaic=2,
    optimizer="AdamW",
    lr0=2e-3,
    warmup_bias_lr=0.0,
    weight_decay=0.025,
    momentum=0.9,
    workers=4,
    trainer=YOLOESegTrainerFromScratch,
    device="0,1,2,3,4,5,6,7",
    amp=False,
)


```

**Comment by glenn-jocher** [comment]
Thanks for sharing that; `amp=False` does suggest this is mixed-precision instability during YOLOE from-scratch training, so if anyone can still reproduce it on the latest package, please open a new issue with a small reproducible example, `yolo checks`, and the full traceback for visibility, using the [MRE guide](https://docs.ultralytics.com/help/minimum-reproducible-example/).

**Comment by Snitro** [comment]
I didn't change any settings and followed the configuration (including code and datasets) provided in the [official documentation.](https://docs.ultralytics.com/models/yoloe/#launching-training-from-scratch)
My GPU is an NVIDIA H100 80GB HBM3 (81079 MiB), and it may not be suitable for using `amp=True`.

**Comment by ImJaewooChoi** [comment]
@Snitro In fact, when I set `amp=False`, I didn’t encounter that NaN error either. However, the current Ultralytics YOLOE26 uses a different training method than the original YOLOE. Of course, I did encounter NaN values in the original YOLOE as well.

**Comment by Snitro** [comment]
@ImJaewooChoi Good news!
Another noticeable change is that after setting `amp=False`, the training time decreased from about 2 hours per epoch to around 40 minutes per epoch.
